llvm/test/SYCLLowerIR/addrspacecast_handling.ll

```diff
@@ -0,0 +1,83 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; RUN: opt < %s -LowerWGScope -S | FileCheck %s
+
```

Comment:
Looks like SYCLLowerIR part of this patch is a duplicate of #3120, which I reviewed.
Please see review comments there.

Reply:
Right. As I mentioned in [here](https://github.com/intel/llvm/pull/2864#issuecomment-769847755), I just pulled #3120 here to run pre-commit tests. I'll merge with the final version of #3120 once it's committed.